lib/jxl/modular/encoding/enc_encoding.cc

```diff
@@ -26,24 +25,45 @@
 #include "lib/jxl/fields.h"
 #include "lib/jxl/image_ops.h"
 #include "lib/jxl/modular/encoding/context_predict.h"
+#include "lib/jxl/modular/encoding/enc_debug_tree.h"
 #include "lib/jxl/modular/encoding/enc_ma.h"
 #include "lib/jxl/modular/encoding/encoding.h"
 #include "lib/jxl/modular/encoding/ma_common.h"
 #include "lib/jxl/modular/options.h"
 #include "lib/jxl/modular/transform/transform.h"
 #include "lib/jxl/toc.h"
 
-#if JXL_OS_IOS
-#define JXL_ENABLE_DOT 0
-#else
-#define JXL_ENABLE_DOT 1  // iOS lacks C89 system()
-#endif
-
 namespace jxl {
 
 namespace {
 // Plot tree (if enabled) and predictor usage map.
 constexpr bool kWantDebug = false;
+constexpr bool kPrintTree = false;
+
+inline std::array<uint8_t, 3> PredictorColor(Predictor p) {
+  switch (p) {
+    case Predictor::Zero:
+      return {{0, 0, 0}};
+    case Predictor::Left:
+      return {{255, 0, 0}};
+    case Predictor::Top:
+      return {{0, 255, 0}};
+    case Predictor::Average0:
+      return {{0, 0, 255}};
+    case Predictor::Average4:
+      return {{192, 128, 128}};
+    case Predictor::Select:
+      return {{255, 255, 0}};
+    case Predictor::Gradient:
+      return {{255, 0, 255}};
+    case Predictor::Weighted:
+      return {{0, 255, 255}};
+      // TODO
```

Comment:
Might do that while we're at it :) (i.e. pick colors for all predictors)

Reply:
It's probably a good idea to make all code related to producing debug stuff compiled conditionally instead of using constexpr conditions that are false (which requires manually editing the code to get it to actually produce stuff). Maybe we need a new build flag for that - it's a bit orthogonal to debug asserts and symbols etc, since these things are not really about debugging the code but rather about seeing what the encoder is doing. @deymo wdyt?

Reply:
constexpr has the advantage that the code still compiles even if it is optimized away, so it tends to bit rot less. For dump_image in AuxOut the current setup uses dependency injection so the code that dumps the stuff is not in the library (in the dump_image case it was a bit more critical to not have it in the library). I think this PR is fine, the debug stuff is now more contained and easier to move out of the enc lib or disable when not needed. Enc library size is not that big of a concern.